### PR TITLE
Improved rule targeting for inter-file diff scan

### PR DESCRIPTION
**Pull request by ihji**
test plan: https://github.com/semgrep/semgrep-proprietary/pull/1396

### CODE REVIEW

**Review by nmote — APPROVED** [review]
Makes sense. Could you fill out the checklist before merging?

File: src/core/Target.ml [hunk 1]
```diff
@@ -27,10 +27,13 @@ type lockfile = {
 }
```
Comment (IagoAbal) [inline src/core/Target.ml:27]: No changelog.

Comment (IagoAbal) [reply src/core/Target.ml:27]: And what does "improved" mean? PR description says nothing, neither here nor in the Pro PR.

File: src/core/Target.mli [hunk 2]
```diff
@@ -63,12 +63,20 @@ type lockfile = {
 (** A lockfile to be used during matching. See also {!Lockfile_xtarget.t}, an
     augmented version with the contents of the lockfile. *)
 
+(** A scan mode that tells us whether we need to do both Intra-file
+    and Inter-file scans or just an Inter-file scan. When performing
```
Comment (IagoAbal) [inline src/core/Target.mli:67]: When do we need to do *both* an intra-file and an inter-file scan? Maybe it makes sense when I look into the Pro PR, but this data type even with the comment is not very clear to me.

**Review by IagoAbal — CHANGES_REQUESTED** [review]
PR description and code comments are not clear enough.

File: src/core/Target.ml [hunk 3]
```diff
@@ -27,10 +27,13 @@ type lockfile = {
...
 [@@deriving show]
 
+type scan_mode = All | InterOnly [@@deriving show]
+
 type regular = {
   path : path;
   analyzer : Xlang.t;
   products : Semgrep_output_v1_t.product list;
+  mode : scan_mode;
```
Comment (IagoAbal) [inline src/core/Target.ml:36]: I checked the Pro PR. I am not sure we need to change this core data type with something that is so specific to inter-file diff scans, when we could just return a pair of `Target.regular * scan_mode`.

Comment (IagoAbal) [reply src/core/Target.ml:36]: Although I'm not even sure about the solution itself, see comments in the Pro PR.

**Pull request closed without merging**